### ISSUE TITLE
Ability to copy objects between cluster

### ISSUE THREAD

**Issue by sebgoa** [bitnami/cabin#31]
_From @sebgoa on July 10, 2016 11:2_

We should be able to copy an object to another cluster that is configured in the app.

i.e get the json/yaml defining that object and creating said object on another cluster.

_Copied from original issue: skippbox/cabin-sw#6_

**Comment by sebgoa** [comment]
This one is not needed for first release of app.

The way to do this is to make an api call to Podtemplate....then fill in the values with the values of the Pod you want to copy, then create the object.

if we do a deployment or rc, then stick the template in the spec of the rc/d....

would be cool, but might be a bit tricky

**Comment by sebgoa** [comment]
Ok there is a simple way to do this for deployments.

When you run a deployment you can use the `--save-config` option, it will store the json of the object in an annotation:

```
$ kubectl run hostname --image=runseb/hostname --save-config

$ kubectl get deployments hostname -o template --template={{.metadata.annotations}}
map[deployment.kubernetes.io/revision:1 kubectl.kubernetes.io/last-applied-configuration:{"kind":"Deployment","apiVersion":"extensions/v1beta1","metadata":{"name":"hostname","creationTimestamp":null,"labels":{"run":"hostname"}},"spec":{"replicas":1,"selector":{"matchLabels":{"run":"hostname"}},"template":{"metadata":{"creationTimestamp":null,"labels":{"run":"hostname"}},"spec":{"containers":[{"name":"hostname","image":"runseb/hostname:0.1.0","resources":{}}]}},"strategy":{}},"status":{}}
```

The json in `kubectl.kubernetes.io/last-applied-configuration:` can be used to target another cluster.

The workflow would be:

1-check if a deployment as annotations and check if is there is a last-applied-configuration
2-if there is a config, display it, maybe like the history
3-and offer to launch it to another cluster with pop-up like the namespace picker...

that gives us simple migration of apps between cluster.

cc/ @kemcake should be relatively simple.

**Comment by sebgoa** [comment]
I create a deployment in cluster 185.19.31.62 called ghost-issue-6 it has such configuration.

you can see the json of the last-applied-configuration. If we get this and let a user choose another cluster, he/she can basically "copy" over a deployment.

![simulator screen shot oct 20 2016 5 23 28 pm](https://cloud.githubusercontent.com/assets/1886299/19566271/f34bd3ca-96e9-11e6-87c6-3fddfa6f6dde.png)

**Comment by sebgoa** [comment]
_From @kemcake on November 17, 2016 16:52_

Done in #115 💪

**Comment by sebgoa** [comment]
can you push a test flight for thise

**Comment by sebgoa** [comment]
_From @kemcake on November 17, 2016 18:14_

I have to wait for the actual version to be released :/

**Comment by sebgoa** [comment]
On the test flight, when I click on "copy cluster" the app crashes...

**Comment by sebgoa** [comment]
_From @kemcake on November 24, 2016 9:44_

fixed on the 0.3.1 Build 3

**Comment by sebgoa** [comment]
ok tested, it works
